draft-irtf-cfrg-frost.md

```diff
@@ -1031,12 +1029,231 @@ signature_share
 
 # Test Vectors
 
-TODO
+This section contains test vectors for all ciphersuites listed in {{ciphersuites}}.
+All Element and Scalar values are represented in serialized form and encoded in
+hexadecimal strings. Signatures are represented as the concatenation of their
+constituent parts. The input message to be signed is also encoded as a hexadecimal
+string.
+
+Each test vector consists of the following information.
+
+- Configuration: This lists the fixed parameters for the particular instantiation
+  of FROST, including MAX_SIGNERS, THRESHOLD_LIMIT, and NUM_SIGNERS.
+- Group input parameters: This lists the group secret key and shared public key,
+  generated by a trusted dealer as described in {{dep-dealer}}, as well as the
+  input message to be signed. All values are encoded as hexadecimal strings.
+- Signer input parameters: This lists the signing key share for each of the
+  NUM_SIGNERS signers.
+- Round one parameters and outputs: This lists the NUM_SIGNERS participants engaged
+  in the protocol, identified by their integer index, the hiding and binding commitment
+  values produced in {{frost-round-one}}, as well as the resulting group commitment list,
+  encoded as described in {{dep-encoding}}, and group binding factor as computed in {{frost-round-two}}).
+- Round two parameters and outputs: This lists the NUM_SIGNERS participants engaged
+  in the protocol, identified by their integer index, along with their corresponding
+  output signature and commitment share as produced in {{frost-round-two}}.
+- Final output: This lists the aggregate signature as produced in {{frost-aggregation}}.
+
+## FROST(Ed25519, SHA512)
 
-<!--
-List of test vector things to include:
-- Domain separated hash functions
-- Shamir secret sharing
-- Encoding group messages
-- Different (n, t) values, and different subsets
--->
+~~~
+// Configuration information
+MAX_SIGNERS: 3
+THRESHOLD_LIMIT: 2
+NUM_SIGNERS: 2
+
+// Group input parameters
+group_secret_key: 7c1c33d3f5291d85de664833beb1ad469f7fb6025a0ec78b3a7
+90c6e13a98304
+group_public_key: 377a6acb3b9b5f642c5ce355d23cac0568aad0da63c633d59d4
+168bdcbce35af
+message: 74657374
+
+// Signer input parameters
+S1 signer_share: 949dcc590407aae7d388761cddb0c0db6f5627aea8e217f4a033
+f2ec83d93509
+S2 signer_share: ac1e66e012e4364ac9aaa405fcafd370402d9859f7b6685c07ee
+d76bf409e80d
+S3 signer_share: d7cb090a075eb154e82fdb4b3cb507f110040905468bb9c46da8
+bdea643a9a02
+
+// Round one parameters
+participants: 1,2
+commitment_list: 000178e175d15cb5cec1257e0d84d797ba8c3dd9b4c7bc50f3fa
+527c200bcc6c4a954cdad16ae67ac5919159d655b681bd038574383bab423614f8967
+396ee12ca62000288a4e6c3d8353dc3f4aca2e10d10a75fb98d9fbea98981bfb25375
+996c5767c932bbf10c41feb17d41cc6433e69f16cceccc42a00aedf72feb5f44929fd
+f2e2fee26b0dd4af7e749aa1a8ee3c10ae9923f618980772e473f8819a5d4940e0db2
+7ac185f8a0e1d5f84f88bc887fd67b143732c304cc5fa9ad8e6f57f50028a8ff
+group_binding_factor: c4d7668d793ff4c6ec424fb493cdab3ef5b625eefffe775
+71ff28a345e5f700a
```

Comment:
Should this go after the Signer round one outputs?

Reply:
Eh, I don't feel strongly. It could.